micro-rdk/src/common/entry.rs

```diff
@@ -97,10 +97,14 @@ pub async fn serve_inner<S>(
 
     let robot_credentials = app_client.robot_credentials();
 
-    let (cfg_response, cfg_received_datetime) = app_client
-        .get_app_config(Some(network.get_ip()))
-        .await
-        .unwrap();
+    let (cfg_response, cfg_received_datetime) = {
+        let config = app_client.get_app_config(Some(network.get_ip())).await;
+        config.unwrap_or_else(|e| {
+            let config = storage.get_robot_configuration().ok();
+            log::error!("Failed to get_app_config CloudConfig, falling back to cached config if available: {e}");
+            (Box::new(ConfigResponse { config }), None)
```

Comment:
Same comment as before feels very weird to re package a cached config into a ConfigResponse. Would make more sense to return something along the lines of `Option<Box<RobotConfig>>`

Reply:
I need to pass *something* into all the code down the line. The robot config is referenced multiple times in this function as parameters for other calls and the function itself (`serve_inner`) doesn't return a result for me to propogate upwards.

By having an `Option<Box<RobotConfig>>` that means I can have `None` (when the device fails to `get_app_config` && fails to get from storage). 

These later calls all assume a `CloudConfig` is available:
- `from_cloud_config`
- initializing `ConfigMonitor` task
- building the server

Should I refactor all those (in new tickets) to consider Options? Or something else?

Reply:
I agree with this refactor, working with a `ConfigResponse` makes less sense since it only wraps a `RobotConfig`